### PR TITLE
Editorial: Create documentation/aria-idl.md

### LINKED ISSUE #2310: Document why IDL decisions were made in the past
- Should the spec document why some of these IDL decisions were made in the past?

_Originally posted by @cookiecrook in https://github.com/w3c/aria/issues/2177#issuecomment-2259210304_

And in https://github.com/w3c/aria/issues/2177#issuecomment-2259171763

### PR DESCRIPTION
**Pull request by rahimabdi**
Closes #2310

This PR adds a new `aria-idl.md` documentation file that provides general explanation about IDL, how ARIA IDL works and chronology/history of key ARIA IDL decisions.

### CODE REVIEW

File: documentation/aria-idl.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    -   Can ensure that only valid data types are passed/returned from APIs which helps with catching errors
+    -   For example, the [<input> element’s DOM interface](https://html.spec.whatwg.org/#the-input-element) contains a disabled boolean attribute
+-   ***Easier binding***:
+    -   IDL describes a higher-level, “abstract” and language-independent description of APIs
+    -   Can easily generate bindings to bridge concrete implementation with API definition
+-   ***Interoperability***: An interface defined in IDL can be implemented in C++ (by a browser engine) and used by a JavaScript application
+-   ***Documentation***:
+    -   Specifies how web APIs should behave for both browser implementers and developers
+    -   Interfaces are largely self documenting
+-   ***Backwards compatibility***: Can add new properties/methods without breaking backwards-compatibility
```
Comment (keithamus) [inline documentation/aria-idl.md:40]: Perhaps worth capturing the consistency between APIs as a feature of IDL:

```suggestion
-   ***Backwards compatibility***: Can add new properties/methods without breaking backwards-compatibility
-  ***Consistency***: New APIs, such as methods and properties, follow similar forms to existing APIs - conventions and types are heavily reused, creating familiarity interface between APIs.
```

**Review by keithamus — APPROVED** [review]
This LGTM overall. Some minor nits;

**Comment by rahimabdi** [comment]
> This LGTM overall. Some minor nits;

Thank you @keithamus! I've incorporated your recommended changes.

**Review by spectranaut — APPROVED** [review]
Awesome.

**Pull request merged**